ENIGMAsystem/SHELL/Audio_Systems/SDL/SDLsystem.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include <map>
+#include <vector>
+#include <SDL.h>
+#include "Widget_Systems/widgets_mandatory.h"
+#include "SDLsystem.h"
+#include "SoundResource.h"
+
+AssetArray<Sound> sounds;
+namespace enigma {
+int audiosystem_initialize() {
+  SDL_Init(SDL_INIT_AUDIO);
+  int flags=MIX_INIT_OGG|MIX_INIT_MOD|MIX_INIT_FLAC|MIX_INIT_MP3;
+  int initted=Mix_Init(flags);
+  if ((initted&flags) != flags) {
+    DEBUG_MESSAGE(std::string("Mix_Init: %s\n")+ std::string(Mix_GetError()),MESSAGE_TYPE::M_ERROR);
+  }
+  if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 4096)==-1) { return 0; }
+  return 1;
+}
+
+void audiosystem_update(void) {}
+
+void audiosystem_cleanup() {
+  Mix_CloseAudio();
+  Mix_Quit();
+}
+
+int sound_add_from_buffer(int id, void* buffer, size_t bufsize) {
+  Sound snd;
+  SDL_RWops* rw = SDL_RWFromMem(buffer, bufsize);
+  snd.mc=Mix_LoadWAV_RW(rw, true);
+  if (!snd.mc) {
+    DEBUG_MESSAGE( "Failed to load the sample", MESSAGE_TYPE::M_ERROR);

```

Comment:
Add the id to the message so the sound can be tracked down.

Reply:
resolved